### ISSUE TITLE
Change npm script production -> prod for yarn

### ISSUE THREAD

**Issue by oscarmorrison** [JeffreyWay/laravel-mix#598]
### Description:
Some of us choose to use `yarn` instead of npm. Currently in `yarn <0.19` running 
`yarn run production` is reserved/broken see [here](https://github.com/yarnpkg/yarn/issues/2349)

I propose that we change the default key to `prod` e.g:

```
"prod": "cross-env NODE_ENV=production node_modules/webpack/bin/webpack.js --progress --hide-modules --config=node_modules/laravel-mix/setup/webpack.config.js"
```

#### Why?
We are already using the shortened work `dev` for development. 
For consistency, and cross compatibility lets use `prod` for production.

### Steps To Reproduce:

- Install `yarn <0.19`
- Setup fresh Laravel build (> 5.4)
- run: `yarn run production`
(you will now see an error)

**Comment by JeffreyWay** [comment]
Seems like they fixed it though, right? 

Either way, you own your package.json file, so you can freely change it from "production" to "prod".

**Comment by oscarmorrison** [comment]
Thanks for the quick response @JeffreyWay 
Yeah they did fix it for 0.19 onward. I agree about owning your own package.json, 
but don't you see the consistency of naming. Why is one dev not development etc?
